lib/graphql/query/serial_execution/selection_resolution.rb

```diff
@@ -6,12 +6,19 @@ def self.resolve(target, current_type, selection, query_ctx)
           selection_result = {}
 
           selection.each_selection(type: current_type) do |name, subselection|
-            selection_result.merge!(query_ctx.execution_strategy.field_resolution.new(
+            field_result = query_ctx.execution_strategy.field_resolution.new(
               subselection,
               current_type,
               target,
               query_ctx
-            ).result)
+            ).result
+
+            if field_result.values[0] == GraphQL::Execution::Execute::PROPAGATE_NULL
```

Comment:
The problem mentioned in https://github.com/rmosolgo/graphql-ruby/pull/416#issuecomment-263365631 is likely related to this check assuming that the selection has been resolved, but in graphql-batch it lazily resolves fields so would only have the promise value as a placeholder at this point.  Instead, get_finished_value should remain the place to check for nulls to propagate.